### DISCUSSION TITLE
pattern matching like pmatch?

### DISCUSSION THREAD

**Discussion by furesoft** [rzubek/CSLisp#23] [Ideas]

**Comment by rzubek** [comment]
I'd love to hear more about the implementation idea - example syntax, how it would work, etc.

**Reply by furesoft** [reply]
here a reference implementation: https://github.com/thezerobit/cl-pmatch

**Reply by rzubek** [reply]
I see what you mean. Yes, porting that to CSLisp would be cool.

We could also consider unification: e.g. https://norvig.com/unify-bug.pdf

It's a way to do matching in a way that ensures consistent variable assignment across the entire pattern, for example `(unify '(X Y is X) '(fast car is fast))` would result in bindings `((X fast) (Y car))` but `(unify `(X Y is X) `(green car is fast))` would fail because X needs to match in both places.

An implementation that pulls the best of both worlds (regex-like patterns from match, and constraint satisfaction from unification) would be really interesting.

**Reply by furesoft** [reply]
Is the implementation planned? Would be really nice
